docs/how_to_contribute/math/how_to_add_prime_fields.md

```diff
@@ -8,7 +8,22 @@ _Note_: We have our own development conventions. Please read the [conventions do
 
 Begin by creating a directory named `<new_prime_field>_prime` in `/tachyon/math/finite_field/`. Add a `BUILD.bazel` file into this directory. Note that once parameters are added to `BUILD.bazel`, Bazel will automatically generate the prime field code based on these parameters when it builds the target.
 
-Choose among `generate_prime_fields()`, `generate_fft_prime_fields()`, or `generate_large_fft_prime_fields()` for code generation, depending on the prime field type. For more information, refer to [`prime_field_generator`](/tachyon/math/finite_fields/generator/prime_field_generator/build_defs.bzl).
+### [Prime field generator](/tachyon/math/finite_fields/generator/prime_field_generator/build_defs.bzl)
+
+Choose the prime field generator depending on the prime field type. The list of required properties for each prime field generator is included below. Note that each successive field type in this list additionally requires all properties mentioned in previous generator types.
+
+1. **`generate_prime_fields()`**:
+    - `name`: The name of the prime field.
+    - `namespace`: The selected namespace, commonly set as `tachyon::math`.
+    - `class_name`: The name of the class, usually the same as the name of the prime field (in PascalCase).
+    - `modulus`: The modulus value of the prime field.
+    - `flag`: The name of the flag exposed in the prime field's configuration (`kIs<PrimeFieldName>`).
+2. **`generate_fft_prime_fields()`**:
+    - `subgroup_generator`: The value used to generate elements of a subgroup within the prime field, facilitating Fast Fourier Transform (FFT) operations.
+      - _Note_: Every prime can be expressed in the following form: $p = 2^s * T + 1$, where $s$ and $T$ are integers, and $T$ is odd. According to Fermat's little Theorem, for any element $a$ in $Fp$ holds that $a^{p-1} = 1 \mod p$. That is to say, $a^{2^s * T} = 1 \mod p$. A subgroup generator $g$ satisfies $g^{2^{s-n} * T} = 1 \mod p$ for some $n < s$, where $2^n$ is the highest power of 2 dividing $2^s * T$, making $g$ a generator of a subgroup of order $2^n$.
+3. **`generate_large_fft_prime_fields()`**:
+    - `small_subgroup_base`: Refers to the base element of a small subgroup.
+    - `small_subgroup_adicity`: Describes the adicity (the power of the base) of a small subgroup within the field.
```

Comment:
Please add note  not only about `subgroup_generator` but also about `small_subgroup_base` and `small_subgroup_adicity`, too